### ISSUE TITLE
False accept: a bare (n), (b) or (P) subscript flag is accepted, which native Zsh rejects

### ISSUE THREAD

**Issue by ss-o** [z-shell/zsh-lint#390]
A subscript pattern flag that requires an argument — `(n)`, `(b)`, `(P)` — is accepted when written bare, although native Zsh rejects it with `bad math expression: operand expected`. Found while measuring #389 on merged `main` (`bd1b087`); independent of that issue and of #387.

### Minimized script

```zsh
#!/usr/bin/env zsh
local -a v=( 10 20 30 )
print ${v[(n)abc]}
```

`zsh -f -n` accepts this; `zsh -f` reports `bad math expression: operator expected at 'abc'`. `zsh-lint` accepts it silently.

### The rule

`(n:expr:)`, `(b:expr:)` and `(P:expr:)` take an argument in the colon form. Written bare, the following pattern text is read as the missing argument's arithmetic expression, which then fails. The argument-bearing spelling is valid and correctly accepted, so this is specifically about the bare form:

| Source | Native | zsh-lint |
| --- | --- | --- |
| `print ${v[(n)abc]}` | run: `bad math expression` | accepted |
| `print ${v[(b)abc]}` | run: `bad math expression` | accepted |
| `print ${v[(P)abc]}` | run: `bad math expression` | accepted |
| `print ${v[(n:2:)abc]}` | valid | accepted |
| `print ${v[(r)abc]}` | valid | accepted |

Unlike #387 and #389, the verdict does **not** depend on whether a range comma is present: the bare flag fails identically with no range, in a first range endpoint, and in a second range endpoint. All nine combinations (3 flags × 3 positions) are false accepts.

### Both oracles

Every row passes `zsh -f -n` and fails only at evaluation, so `zsh -n` alone cannot see this family — the runtime oracle is what exposes it. zsh 5.9.2, `mvdan.cc/sh/v3` v3.14.1, `main` at `bd1b087`, lint verdict from `cmd/zsh-lint-survey`.

### Decidability: partial, and the limit matters

The verdict depends on the variable's declared **type**, which the front end cannot see. For an associative array the same source is valid:

| Declared type of `v` | `print ${v[(n)abc]}` |
| --- | --- |
| `local -a` (array) | `bad math expression` |
| scalar (`v=hello`) | `bad math expression` |
| undeclared | `bad math expression` |
| `typeset -A` (assoc) | **valid** |

So a blanket source-only rejection would false-reject valid associative code, which the workflow's contract forbids as firmly as a false accept. This is the same wall as #389.

It is, however, **less** blocked than #389: three of the four declaration states reject, and the associative case is the lone exception. If `internal/scope` is extended to distinguish associative from normal arrays (it currently tracks only `KindVariable` / `KindFunction` / `KindAlias`), a type-aware rule could cover the array, scalar and undeclared cases — which is most real code — while leaving associative subscripts alone.

Recording it so the family is not rediscovered a third time. Not proposing a parser change.

### Rows that must keep parsing

All valid and currently accepted.

| Source | Native | Current |
| --- | --- | --- |
| `print ${v[(n:2:)abc]}` | valid | accepted |
| `print ${v[(r)abc]}` | valid | accepted |
| `typeset -A v=( a 1 ); print ${v[(n)abc]}` | valid | accepted |

### Provenance

Measured while investigating #389, whose body flagged these flags as "adjacent, probably separate". Confirmed separate: they are not a range rule. Probe varied flag × endpoint position × declaration type, judging each row by `zsh -f -n` **and** a `zsh -f` run.

Zsh manual: `zshparam`, *Subscript Flags*.

**Comment by ss-o** [comment]
Blocked on #391.

This needs associative-vs-normal array tracking in `internal/scope`, which does not exist yet: the native verdict here depends on the subscripted variable's declared type, and `internal/parse` cannot see it. A source-only rule would false-reject valid associative code.

#391 is the enabler (the AST already carries the `-A` flag; `internal/scope` just does not record it). Once it lands, the rule for this issue belongs in an analyzer rule reading the symbol map, not in the parser front end.
